develocity-gradle-plugin-adapters/src/compatibilityApi/java/com/gradle/develocity/agent/gradle/adapters/internal/ProxyFactory.java

```diff
@@ -14,7 +14,11 @@
 public final class ProxyFactory {
 
     public static <T> T createProxy(Object target, Class<T> targetInterface) {
-        return newProxyInstance(targetInterface, new ProxyingInvocationHandler(target));
+        return createProxy(target, targetInterface, target.getClass().getClassLoader());
+    }
+
+    private static <T> T createProxy(Object target, Class<T> targetInterface, ClassLoader develocityTypesClassLoader) {
```

Comment:
💬 Prior to this change, we used the local classloader to create nested proxies for return values. For example, for a method like `Property set(TargetEnum)` on `Target` class, we would adapt the args using the class loader of the target object. Then, we would create a proxy to the return value using the classloader of `Property`. However, we would not be able to load any of target classes using it. 

To work around this issue, I pass the top-level classloader containing Develocity classes to all nested proxies under assumption that all other classes like Gradle's `Property` would also be accessible from it.